internal/store/persistentvolume.go

```diff
@@ -102,11 +102,32 @@ var (
 			Type: metric.Gauge,
 			Help: "Information about persistentvolume.",
 			GenerateFunc: wrapPersistentVolumeFunc(func(p *v1.PersistentVolume) *metric.Family {
+				var volLabelIDKey, volLabelIDValue string
+				switch {
+				case p.Spec.PersistentVolumeSource.GCEPersistentDisk != nil:
+					volLabelIDKey = "gce_persistent_disk_name"
+					volLabelIDValue = p.Spec.PersistentVolumeSource.GCEPersistentDisk.PDName
+				case p.Spec.PersistentVolumeSource.AWSElasticBlockStore != nil:
+					volLabelIDKey = "ebs_volume_id"
+					volLabelIDValue = p.Spec.PersistentVolumeSource.AWSElasticBlockStore.VolumeID
+				}
+
+				if volLabelIDKey == "" {
```

Comment:
Prometheus will drop the empty value anyways, so why the extra branch?

Reply:
:+1: I'm not sure about empty keys, so I default key to `default`